app/src/main/java/com/hedvig/app/util/apollo/SafeQuery.kt

```diff
@@ -65,7 +65,12 @@ fun <D : Query.Data> ApolloCall<D>.safeWatch(): Flow<QueryResult<D>> {
 private fun <D : Operation.Data> ApolloResponse<D>.toQueryResult(): QueryResult<D> {
   val data = data
   return when {
-    hasErrors() -> QueryResult.Error.QueryError(errors?.first()?.message)
+    hasErrors() -> {
+      val exception1 = errors?.first()?.extensions?.get("exception")
```

Comment:
Not very pretty, but it works

Reply:
Yeah but if we *know* we're getting good error messages inside the .exception.body.message then this is completely fine.
If anything I'd replace exception1 -> exception and instead of doing LinkedHashMap<*, *> I'd do Map<*, *> in both cases, but that's it. It's also quite safe in the sense that we're always going for a nullable type for the entire chain. So go for it imo!
We'd just have to check that it does not sometimes give off super janky responses or things we shouldn't show. In general it shouldn't be the case though I think